include/ideep/operators/deconv.hpp

```diff
@@ -670,15 +663,34 @@ struct convolution_transpose_forward : public dnnl::deconvolution_forward {
                   strides, dil_compatible, padding_l, padding_r, op_attr, aalgorithm,
                   aprop_kind, aengine);
 
-    tensor src_zero_point_m;
-    conv_deconv_utils::obtain_runtime_zero_point(src, src_zero_point, DNNL_ARG_SRC,
-                                                 op_attr, aengine, src_zero_point_m);
     param.primitive = std::move(super(param.pd));
+    param.op_attr = std::move(op_attr);
     param.bias_attr = std::move(bias_attr);
     param.groups = groups;
-    param.sq_param_ptr =
-        std::make_shared<deconv_forward_quant_params>(std::move(src_zero_point_m));
-    IDEEP_ENFORCE(param.sq_param_ptr, "Failed to allocate memory for quantization parameters");
+
+    // Prepare tensors for scales and zero points
+    if (param.op_attr.has_scales()) {
+      if (!param.all_scales) {
+        param.all_scales.reset(new std::unordered_map<int, tensor>);
+      }
+      for (auto& arg_scale_pair : param.op_attr.get_all_scales()) {
```

Comment:
Why can't we use `param.op_attr.get_all_scales()` directly in the execution?

Reply:
> Why can't we use `param.op_attr.get_all_scales()` directly in the execution?

Because `op_attr` stores scales and zero points as `vector`s but execution needs `tensor`s. So, here we construct `tensor`s and put them in `forward_params`.

Reply:
Would there be copy overhead for per-channel quant scales?

Reply:
Yes, if we want to keep the `const` semantics. It happens on preparation stage, so it won't exist if we have cache. The copying seems inevitable, and I haven't thought of a good idea yet.